h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -137,18 +136,48 @@ class SearchBarController extends Controller {
     };
 
     /**
-     * Updates the hidden input field with the consolidated
-     * search terms from the lozenges.
+     * Insert a hidden <input> into the search <form>.
+     *
+     * The value of the hidden <input> is a search string constructed from
+     * the lozenges plus any text currently in the visible <input>.
+     *
+     * The name="q" attribute is moved from the visible <input> on to the
+     * hidden <input> so that when the <form> is submitted it's the value of
+     * the _hidden_ input, not the visible one, that is submitted as the
+     * q parameter.
+     *
      */
-    var updateHiddenInputValue = () => {
-      var newValue = '';
-      Array.from(this._lozengeContainer.querySelectorAll('.js-lozenge__content'), function(loz) {
-        newValue = newValue + loz.textContent + ' ';
+    var insertHiddenInput = () => {
+      var hiddenInput = document.createElement('input');
+      hiddenInput.type = 'hidden';
+
+      Array.from(this._lozengeContainer.querySelectorAll('.js-lozenge__content')).forEach((loz) => {
```

Comment:
Stylistic nit. Arrow functions with a single argument should omit braces around the argument.